dev/benchmarks/platform_channels_benchmarks/lib/main.dart

```diff
@@ -0,0 +1,174 @@
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:microbenchmarks/common.dart';
+
+List<Object> _makeTestBuffer(int size) {
+  final List<Object> answer = <Object>[];
+  for (int i = 0; i < size; ++i) {
+    switch (i % 9) {
+      case 0:
+        answer.add(1);
+        break;
+      case 1:
+        answer.add(math.pow(2, 65));
+        break;
+      case 2:
+        answer.add(1234.0);
+        break;
+      case 3:
+        answer.add(null);
+        break;
+      case 4:
+        answer.add(<int>[1234]);
+        break;
+      case 5:
+        answer.add(<String, int>{'hello': 1234});
+        break;
+      case 6:
+        answer.add('this is a test');
+        break;
+      case 7:
+        answer.add(true);
+        break;
+      case 8:
+        answer.add(Uint8List(64));
+        break;
+    }
+  }
+  return answer;
+}
+
+Future<void> _runTests() async {
+  assert(false,
+      "Don't run benchmarks in checked mode! Use 'flutter run --release'.");
+  const int numMessages = 10000;
+
+  const BasicMessageChannel<Object> channel = BasicMessageChannel<Object>(
+      'dev.flutter.echo.basic.standard', StandardMessageCodec());
+  final Stopwatch watch = Stopwatch();
+
+  watch.start();
+  for (int i = 0; i < numMessages; ++i) {
+    await channel.send(1234);
+  }
+  watch.stop();
+  final double smallPayloadTime = watch.elapsedMicroseconds / numMessages;
+
+  watch.reset();
+  final List<Object> largeBuffer = _makeTestBuffer(1000);
+  const StandardMessageCodec codec = StandardMessageCodec();
+  final ByteData data = codec.encodeMessage(largeBuffer);
+  print('Large buffer size: ${data.lengthInBytes}');
+  int size = 0;
+  watch.start();
+  for (int i = 0; i < numMessages; ++i) {
+    final List<Object> result = await channel.send(largeBuffer) as List<Object>;
+    // This check should be tiny compared to the actual channel send/receive.
+    size += (result == null) ? 0 : result.length;
+  }
+  watch.stop();
+  final double largePayloadTime = watch.elapsedMicroseconds / numMessages;
+
+  if (size != largeBuffer.length * numMessages) {
+    throw Exception(
+        'There is an error with the echo channel, the results don\'t add up: $size');
+  }
+
+  final BenchmarkResultPrinter printer = BenchmarkResultPrinter();
+  printer.addResult(
+    description:
+        '  BasicMessageChannel/StandardMessageCodec/Flutter->Host/Small',
+    value: smallPayloadTime,
+    unit: 'µs',
+    name: 'platform_channel_basic_standard_2host_small',
+  );
+  printer.addResult(
+    description:
+        '  BasicMessageChannel/StandardMessageCodec/Flutter->Host/Large',
+    value: largePayloadTime,
+    unit: 'µs',
+    name: 'platform_channel_basic_standard_2host_large',
+  );
+  printer.printToStdout();
+}
+
+void main() {
+  runApp(const MyApp());
```

Comment:
Do you actually need the Counter app UI? To simplify, could this just be:

```dart
void main() {
  WidgetsFlutterBinding.ensureInitialized();
   _runTests();
}
```

Reply:
That doesn't work because that doesn't setup the binary messenger for platform channels.  I've removed most of the UI.

Reply:
That's supposed to set it up though, right?

If not, can we just directly do what's needed to set up the default binary messenger?

Reply:
It doesn't say anything about that in the documentation.  What's the concern?  My understanding is that goderbauer didn't want all this extra code, it's mostly been removed.  If we had a `PlatformChannelsBinding.ensureInitialized()` function call it would save  maybe 6 lines of code if I paired down the UI even further.  I imagine things might get a bit goofy if you don't present any widget though.  Do you want me to trim even more away?

Reply:
For me it's more that it takes that much longer to find out what this test is actually doing from here. I have to read through a bunch of unrelated widgets to finally get to one that calls a method in `initState` that has nothing to do with the widget itself. I also have to think a little bit about whether that could have any impact on the benchmark.

I'm kind of surprised the WidgetsFlutterBinding.ensureInitialized isn't enough to get the channels set up - that's what we tell people to do if they need to talk to platform channels before any UI is rendered after all.

This test doesn't really need any UI to run - it'd just show a black screen in that case. If that bothers you, you can just have it show a `Placeholder` widget, or maybe some text saying it's running the benchmark.

Reply:
And in case it's not clear, IMO this shouldn't block the test from landing. But future us will probably be grateful for any simplifications that can be made now :)

Reply:
I think we should remove everything from the benchmark that's not required for it to make it clear what this is testing. Since this benchmark doesn't rely on any UI, why show it? If you really need to show UI, then just show a simple container.

Reply:
Okay, I removed everything I could.  It just shows a black screen now while executing.